### DISCUSSION TITLE
[FEATURE] "schedule.add" on Master; schedule module only on minion event bus

### DISCUSSION THREAD

**Discussion by eliasp** [saltstack/salt#67565] [Feature Requests]
**Description**
Trying to add a scheduled job to the Master (e.g. a scheduled Orchestration) fails.
The `schedule.add` function executed by the `salt.cmd` runner just returns `False` and the comment "Failed to add job test0815 to schedule.".
Listing scheduled jobs also shows no change.

**Setup**
- have a Master running
- have a shell ready to execute `salt-run`

**Steps to Reproduce the behavior**
- Execute on the Master:
  ```bash
  salt-run salt.cmd schedule.add schedule0815 function="test.ping" once="2025-01-01 15:30:00" persist=False
  ```

**Expected behavior**
The execution should return `True` instead of `False` and a corresponding job should show up in `salt-run salt.cmd salt.list`

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>

```
Salt Version:
          Salt: 3003

Dependency Versions:
          cffi: Not Installed
      cherrypy: 8.9.1
      dateutil: 2.7.3
     docker-py: Not Installed
         gitdb: 2.0.6
     gitpython: 3.0.7
        Jinja2: 2.10.1
       libgit2: 0.28.3
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.6.1
        pygit2: 1.0.3
        Python: 3.8.5 (default, Jan 27 2021, 15:41:15)
  python-gnupg: 0.4.5
        PyYAML: 5.3.1
         PyZMQ: 18.1.1
         smmap: 2.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.2

System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 4.15.0-124-generic
        system: Linux
       version: Ubuntu 20.04 focal
```
</details>

**Additional context**
I created an [integration test](https://github.com/eliasp/salt/blob/master-schedule-add/tests/integration/runners/test_salt.py#L50) which demonstrates this issue.

I also tried hunting down the root of the problem, but didn't really succeed so far - it seems to be related to the unsuccessful [execution of `event.fire`](https://github.com/saltstack/salt/blob/master/salt/modules/schedule.py#L488) which results in `add()` returning `ret` containing the default `result` value of `False`.

**Comment by garethgreenaway** [comment]
@eliasp Thanks for the report.  The major issue here is that the schedule module utilizes the event bus specifically the minion event bus and wasn't designed to work with the Master.  When the module runs via the `salt.cmd` runner it's attempting to connect to a minion event bus which doesn't exist.
